apps/bored/bored.star

```diff
@@ -20,26 +20,36 @@ DEFAULT_FONT = "6x13"
 DEFAULT_DIRECTION = "vertical"
 
 def main(config):
-    activity = cache.get("activity")
+    # Create unique cache key based on config values.
+    # The only one that really matters for now is the number of participants
+    friends = config.get("friends", "1")
+
+    # if a random is selected, randomize
+    if friends == "random":
+        friends = str(random.number(1, 5))
+
+    cache_key = "bored_app_" + friends
+
+    activity = cache.get(cache_key)
     if activity != None:
         print("Hit! Displaying cached data.")
     else:
         print("Miss! Calling Bored API.")
 
-        friends = config.get("friends", "1")
-        if friends == "random":
-            friends = random.number(1, 5)
-
         # this may not work
         params = {
-            "participants": str(friends),
+            "participants": friends,
         }
+
         print("params", params)
         rep = http.get(BORED_URL, params = params)
         if rep.status_code != 200:
+            # if the APi fails, return [] to skip this app showing
             fail("Bored request failed with status %d", rep.status_code)
+            return []
```

Comment:
Calling `fail()` will immediately halt execution, so the `return []` won't be reached.